pkg/util/nsenter/nsenter.go

```diff
@@ -64,43 +65,53 @@ const (
 type Nsenter struct {
 	// a map of commands to their paths on the host filesystem
 	paths map[string]string
+	// directories to search for required commands
+	searchDirs []string
+
+	hasSystemd *bool
 }
 
 // NewNsenter constructs a new instance of Nsenter
 func NewNsenter() (*Nsenter, error) {
 	ne := &Nsenter{
 		paths: map[string]string{
-			"mount":       "",
-			"findmnt":     "",
-			"umount":      "",
-			"systemd-run": "",
-			"stat":        "",
-			"touch":       "",
-			"mkdir":       "",
-			"ls":          "",
-			"sh":          "",
-			"chmod":       "",
+			"mount":   "",
+			"findmnt": "",
+			"umount":  "",
+			"stat":    "",
+			"touch":   "",
+			"mkdir":   "",
+			"ls":      "",
+			"sh":      "",
+			"chmod":   "",
+			"nsenter": "",
 		},
+		searchDirs: []string{"/bin", "/usr/sbin", "/usr/bin", "/"},
 	}
-	// search for the required commands in other locations besides /usr/bin
 	for binary := range ne.paths {
-		// check for binary under the following directories
-		for _, path := range []string{"/", "/bin", "/usr/sbin", "/usr/bin"} {
-			binPath := filepath.Join(path, binary)
-			if _, err := os.Stat(filepath.Join(hostRootFsPath, binPath)); err != nil {
-				continue
-			}
-			ne.paths[binary] = binPath
-			break
-		}
-		// systemd-run is optional, bailout if we don't find any of the other binaries
-		if ne.paths[binary] == "" && binary != "systemd-run" {
-			return nil, fmt.Errorf("unable to find %v", binary)
+		if found := ne.findBinary(binary); !found {
+			return nil, fmt.Errorf("unable to find binary %s for nsenter in any of %s", binary, strings.Join(ne.searchDirs, ", "))
 		}
 	}
 	return ne, nil
 }
 
+// findBinary searches for the executable binary path
+func (ne *Nsenter) findBinary(binary string) bool {
+	var found bool
+	for _, path := range ne.searchDirs {
+		binPath := filepath.Join(path, binary)
+		if _, err := os.Stat(filepath.Join(hostRootFsPath, binPath)); err != nil {
```

Comment:
instead of checking for `err != nil` can we check for `err == nil` so we can set the `ne.paths[binary]` within the braces and return true?

(avoids the extra `found` and breaking out of the loop. we just return `false` at the end of the method